### PR TITLE
adding meta library

### PR DESCRIPTION
**Pull request by ishidawataru**
This PR adds TAI meta library which is a helper library for TAI to provide serialization/validation functions.

Some portion of the code will be auto-generated by python script, main.py.
It internally uses clang to parse TAI header files.

This PR also adds a commit which uses this meta library for taish command parsing.

### CODE REVIEW

**Comment by ishidawataru** [comment]
@scott-emery @iTetsuyaMurakami PTAL

**Comment by ishidawataru** [comment]
@scott-emery Do you have any comment on this? Since Travis is now working, I want to add test/format checker using this meta library after this get merged. Then we can add various attributes we discussed in the TAI workshop safely.

File: inc/tainetworkif.h [hunk 1]
```diff
@@ -60,36 +60,23 @@ typedef enum _tai_network_interface_oper_status_t
...
-    TAI_NETWORK_INTERFACE_TX_ALIGN_OUT              = 0x02,
-    TAI_NETWORK_INTERFACE_TX_ALIGN_CMU_LOCK         = 0x04,
-    TAI_NETWORK_INTERFACE_TX_ALIGN_REF_CLOCK        = 0x08,
-    TAI_NETWORK_INTERFACE_TX_ALIGN_TIMING           = 0x10
+    TAI_NETWORK_INTERFACE_TX_ALIGN_LOSS,
+    TAI_NETWORK_INTERFACE_TX_ALIGN_OUT,
+    TAI_NETWORK_INTERFACE_TX_ALIGN_CMU_LOCK,
+    TAI_NETWORK_INTERFACE_TX_ALIGN_REF_CLOCK,
+    TAI_NETWORK_INTERFACE_TX_ALIGN_TIMING
 } tai_network_interface_tx_align_status_t;
```
Comment (scott-emery) [inline inc/tainetworkif.h:68]: The reason these were defined as bitmaps was that there could be multiple reasons for the transmitter to be out of alignment. So TX_ALIGN_OUT and TX_ALIGN_REF_CLOCK could both be set. A bitmap seemed like a reasonable way to represent this. Is there any reason why you changed this to an enum with incrementing values and modified the attribute to be a list?

File: inc/taihostif.h [hunk 2]
```diff
@@ -87,15 +87,15 @@ typedef enum _tai_host_interface_attr_t
      *
      * A list of lane fault status
      *
-     * @type #tai_u32_list_t #tai_host_interface_lane_faults_t
+     * @type #tai_s32_list_t #tai_host_interface_lane_fault_t
      * @flags READ_ONLY
      */
-    TAI_HOST_INTERFACE_ATTR_LANE_FAULTS,
```
Comment (scott-emery) [inline inc/taihostif.h:93]: Since this is a list, the plural "FAULTS" seems appropriate. What was the reason for this change?

**Review by scott-emery — COMMENTED** [review]
Some of the changes to the TAI header files seem unnecessary. Can you provide more explanation on why the changes were made?

File: inc/tainetworkif.h (hunk 1 above)
Comment (ishidawataru) [reply inc/tainetworkif.h:68]: https://github.com/opencomputeproject/SAI/blob/master/inc/saiport.h#L275-L281

I'm following SAI. This also makes implementing meta library simple.

File: inc/taihostif.h (hunk 2 above)
Comment (ishidawataru) [reply inc/taihostif.h:93]: Same reason as above.

`SAI_PORT_ATTR_SUPPORTED_FEC_MODE` is an enum list but it is not `SAI_PORT_ATTR_SUPPORTED_FEC_MODES`

I think it's clearer to make the attr name similar to the attr type name which is `tai_host_interface_lane_fault_t` in this case.

Comment (scott-emery) [reply inc/taihostif.h:93]: Ok, I'm fine with this. At some point we need to be a little more restrictive when changing the names of attributes. Every time we do this, all of the code (tai adapters and tai applications) must be modified to use the new names.

Comment (ishidawataru) [reply inc/taihostif.h:93]: Yes, I agree. At this point, I don't have any more proposal for the naming change.

**Pull request merged**